### PR TITLE
Optimize the MockInvokersSelector

### PR DESCRIPTION
**Pull request by pinxiong**
## What is the purpose of the change

Optimize the MockInvokersSelector

## Brief changelog

1. Remove the unused import
2. Use CollectionUtils#isEmptyMap to check invocation's attachments
3. Remove the unused MockInvokersSelector#hasMockProviders to improve the performance
4. Add MockInvokersSelector#isMockProtocol to check if the given invoker is mock protocol

## Verifying this change

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by pinxiong** [comment]
@chickenlj

File: dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/mock/MockInvokersSelector.java [hunk 1]
```diff
@@ -63,41 +63,31 @@ public MockInvokersSelector() {
...
+            if (isMockProtocol(invoker)) {
                 sInvokers.add(invoker);
             }
         }
         return sInvokers;
     }
 
     private <T> List<Invoker<T>> getNormalInvokers(final List<Invoker<T>> invokers) {
-        if (!hasMockProviders(invokers)) {
-            return invokers;
```
Comment (chickenlj) [inline dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/mock/MockInvokersSelector.java:80]: On most occasions, the call will hit this if branch and return directly.  I don't think the new way will bring any performance impromement.

**Review by chickenlj — CHANGES_REQUESTED** [review]

File: dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/mock/MockInvokersSelector.java (hunk 1 above)
Comment (pinxiong) [reply dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/mock/MockInvokersSelector.java:80]: > On most occasions, the call will hit this if branch and return directly. I don't think the new way will bring any performance impromement.

There are no obvious differences of performance between the previous way and the new way. However, the simple changes can make the code have better reusability and readability.

Comment (chickenlj) [reply dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/mock/MockInvokersSelector.java:80]: I worry there'll have a performance drop for allocating Arraylist for every call.

Comment (pinxiong) [reply dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/mock/MockInvokersSelector.java:80]: You're right, thanks!

**Pull request closed without merging**